src/hooks/restrict-to-owner.js

```diff
@@ -0,0 +1,42 @@
+import errors from 'feathers-errors';
+
+const defaults = {
+  idField: '_id',
+  ownerField: 'userId'
+};
+
+export default function(options = {}){
+  return function(hook) {
+    if (hook.type !== 'before') {
```

Comment:
Let's create a new issue for this and do it in a new PR. This one is already way too massive, and it won't be a breaking change.